inffast.c

```diff
@@ -107,10 +108,11 @@ void ZLIB_INTERNAL inflate_fast(z_stream *strm, unsigned long start) {
        input data or output space */
     do {
         if (bits < 15) {
-            hold += (uint32_t)(*in++) << bits;
-            bits += 8;
-            hold += (uint32_t)(*in++) << bits;
-            bits += 8;
+            unsigned short chunk;
+            __builtin_memcpy(&chunk, in, sizeof(unsigned short));
+            hold += (chunk << bits);
```

Comment:
This hunk with the other similar one below is the only change that is unsafe for big-endian machines.
make check passes on powerpc-linux with these two hunks removed.